platform/default/mbgl/storage/offline_database.cpp

```diff
@@ -877,19 +877,27 @@ bool OfflineDatabase::evict() {
   stmt1->run();
   
   // clang-format off
-  Statement stmt2 = getStatement(
+  while (true) {
+    Statement stmt2 = getStatement(
                                  "DELETE FROM tiles "
                                  "WHERE id IN ( "
                                  "  SELECT id FROM tiles "
                                  "  LEFT JOIN region_tiles "
                                  "  ON tile_id = tiles.id "
                                  "  WHERE tile_id IS NULL "
                                  "  AND accessed <= ?1 "
-                                 ") ");
-  // clang-format on
-  stmt2->bind(1, accessed);
-  stmt2->run();
+                                 ") LIMIT 10000");
   
+    // clang-format on
+    stmt2->bind(1, accessed);
+    stmt2->run();
+    Log::Info(Event::Database, "deleted tiles %d", stmt2->changes());
+
+    if (stmt2->changes() < 10000) {
+      break;
+    }
+
+  }
```

Comment:
I don't have strong science behind this change, I thought it would be more stable if this was a series of short running database operations, and less likely to generate an enormous journal file. If it crashes mid operation the large journal files are processed on the next launch, while the loading screen is showing.